### DISCUSSION TITLE
Make model transparent by setting alpha values, change blending mode at runtime.

### DISCUSSION THREAD

**Discussion by MukeshBlippar** [google/filament#6441] [Ideas]
Hi, I have requirement to set Fade in / out animation on 3D models. And also to change the opacity of any model at run time to make it transparent. So far from all the issues here, I learnt that it's only possible if the blending mode of model is set 'Transparent' or 'Fade'. But if the default blending mode is 'Opaque', it is not allowing to set the alpha values. And Filament API also doesn't allow to change the blending mode at run time.

I managed to set opacity for the model by creating my own custom material and applying the on renderable. But this over rides the existing material. So the question is how can we achieve to set the alpha value in existing material. Is it possible to change the blending mode at the app level? Any plan to add this feature in future? Here is my code which is working to create material with alpha value.

```
Material materialFilament;
 MaterialBuilder.init();
        MaterialBuilder materialBuilder = new MaterialBuilder()
                .platform(MaterialBuilder.Platform.MOBILE)
                .name("External Glb Material")
                .require(MaterialBuilder.VertexAttribute.UV0)
                .shading(MaterialBuilder.Shading.UNLIT)
                .doubleSided(false)
                .optimization(MaterialBuilder.Optimization.NONE);

        MaterialPackage materialPackage = materialBuilder
                .blending(MaterialBuilder.BlendingMode.TRANSPARENT)
                .material("void material(inout MaterialInputs material) {\n" +
                        "    prepareMaterial(material);\n" +
                        "    material.baseColor.a = 0.30;\n" +
                        "    material.baseColor.rgb *= material.baseColor.a;\n" +
                        "}\n")
                .build(filamentEngine);

        if (materialPackage.isValid()) {
            ByteBuffer buffer = materialPackage.getBuffer();
            materialFilament = new com.google.android.filament.Material.Builder()
                    .payload(buffer, buffer.remaining())
                    .build(filamentEngine);
        }
        MaterialBuilder.shutdown();
```

**Comment by bejado** [comment]
> Is it possible to change the blending mode at the app level?

No, the blending mode informs how the shader code is generated, which happens offline during material compilation. If you want to be able to control transparency on all materials, you'll either need to change all their blend modes, or maintain two versions of each material.
